app/delegators/delegator_decorator.rb

```diff
@@ -0,0 +1,17 @@
+module DelegatorDecorator
+  def initialize(obj)
+    super obj
+    @delegate_sd_obj = obj
+  end
+
+  def __getobj__
+    @delegate_sd_obj
+  end
+  alias_method :model, :__getobj__
+
+  def __setobj__(obj)
```

Comment:
We aren't really modifying this method from the [Delegator class](http://ruby-doc.org/stdlib-2.3.1/libdoc/delegate/rdoc/Delegator.html) are we?